### PR TITLE
TE : adding anomaly merge rest end points

### PR DESCRIPTION
**Pull request by puneetjaiswal**
Adding anomaly merge rest end points

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalySummaryResource.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      anomalies =
+          resultDAO.findAllByCollectionTimeAndMetric(collection, metric, startTimeUtc, endTimeUtc);
+    } else {
+      anomalies = resultDAO.findAllByCollectionAndTime(collection, startTimeUtc, endTimeUtc);
+    }
+    AnomalyMergeConfig defaultMergeConfig = new AnomalyMergeConfig();
+    return AnomalySummaryGenerator.mergeAnomalies(anomalies, defaultMergeConfig);
+  }
+
+  @GET
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalySummaryResource.java]: What is this function doing? Who will be using this count?
Is it grouping by only function or strategy (should it be names getAnomalyDataGroupedByStrategy) ?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java [hunk 2]
```diff
@@ -35,21 +39,39 @@
...
 
   private static final String FIND_VALID_BY_TIME_EMAIL_ID =
       "SELECT r FROM EmailConfiguration d JOIN d.functions f JOIN f.anomalies r "
-      + "WHERE d.id = :emailId AND ((r.startTimeUtc >= :startTimeUtc AND r.startTimeUtc <= :endTimeUtc) "
-      + "OR (r.endTimeUtc >= :startTimeUtc AND r.endTimeUtc <= :endTimeUtc)) "
+          + "WHERE d.id = :emailId AND ((r.startTimeUtc >= :startTimeUtc AND r.startTimeUtc <= :endTimeUtc) "
+          + "OR (r.endTimeUtc >= :startTimeUtc AND r.endTimeUtc <= :endTimeUtc)) "
           + "AND r.dataMissing=:dataMissing";
 
+  private static final String COUNT_GROUP_BY_FUNCTION = "select count(r.id) as num, r.function.id,"
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java]: minor nit for readability: can we keep this consistent in terms of what is capitalized? (all keywords like SELECT, GROUP BY etc have been capitalized in all other query definitions)

**Comment by npawar** [comment]
LGTM

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalySummaryResource.java (hunk 1 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalySummaryResource.java]: counts for each category will be used on the ui to summarize anomalies between a time selection.
Function is only one of the strategies. Strategy was obvious so I omitted it.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java (hunk 2 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java]: I feel WHEN the WORDS jumbled IN both CAPS and LOWERCASE, it MAKES things HARDER to READ.
Using all lowercase and only only entities camel case is better to read. There is no difference performance wise. Also IDE highlights the sql keywords.

There has been many discussions around this http://stackoverflow.com/questions/608196/why-should-i-capitalize-my-sql-keywords

**Pull request merged**